pkg/config/config.go

```diff
@@ -322,108 +327,99 @@ func NewConfig(nn types.NamespacedName, uid types.UID, currentState *SpiceDBMigr
 	}, warning, nil
 }
 
-func validateImage(image string, globalConfig *OperatorConfig) (string, []error) {
-	if len(image) == 0 && len(globalConfig.ImageName) == 0 {
-		return "", []error{fmt.Errorf("no defualt image configured for operator and no image provided in spec")}
+func computeTargets(image, version, channel, engine string, currentVersion *v1alpha1.SpiceDBVersion, globalConfig *OperatorConfig, rolling bool) (baseImage string, targetSpiceDBVersion *v1alpha1.SpiceDBVersion, state updates.State, err error) {
+	baseImage, tag, digest := ExplodeImage(image)
+
+	// if digest or tag are set, we don't use an update graph
+	if len(digest) > 0 || len(tag) > 0 {
```

Comment:
should we log these different states?

Reply:
It will be visible in the status of the object (it will show an image name only, not a version/channel)

Reply:
I meant more how we compute them based on the incoming data